lib/sql.js

```diff
@@ -1185,8 +1185,22 @@ SQLConnector.prototype.buildOrderBy = function(model, order) {
     return '';
   }
   var self = this;
-  if (typeof order === 'string') {
-    order = [order];
+  var usedConnector = this.dataSource.connector.name;
+
+  if (order === 'rand') {
+    if (usedConnector === 'mysql') {
+      return 'ORDER BY RAND()';
+    } else if (usedConnector === 'postgres') {
+      return 'ORDER BY RANDOM()';
+    } else if (usedConnector === 'oracle') {
+      return 'ORDER BY dbms_random.value';
+    } else if (usedConnector === 'mssql') {
+      return 'ORDER BY NEWID()';
+    }
+  } else {
+    if (typeof order === 'string') {
+      order = [order];
+    }
```

Comment:
In general, I don't like the `if/else` approach here is the base class. We should define an `extension point` in the base class so that individual connectors can contribute their specific implementations.

To support this use case, I would like to introduce system generated/resolved variables in the query object, for example `${random}`. These variables will be substituted at SQL building time with help from connectors.